src/main/java/com/PBA/budgetservice/controller/ExpenseController.java

```diff
@@ -58,4 +59,27 @@ public ResponseEntity<ExpenseDto> updateExpense(@io.swagger.v3.oas.annotations.p
     })
     @GetMapping("/category")
     public ResponseEntity<List<ExpenseCategoryDto>> getAllExpenseCategories();
+
+    @Operation(summary = "Provides a list of all the expenses corresponding to a logged in user, filtered by expense name.")
```

Comment:
Refactor to have only 1 endpoint for filtering
https://pavankjadda.dev/search-data-across-multiple-columns-using-spring-data-jpa/